pkg/launchd/launchd.go

```diff
@@ -0,0 +1,95 @@
+package launchd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"text/template"
+)
+
+const (
+	plistTemplate = `<?xml version='1.0' encoding='UTF-8'?>
+	<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+	<plist version='1.0'>
+		<dict>
+			<key>Label</key>
+			<string>{{ .Label }}</string>
+			<key>ProgramArguments</key>
+			<array>
+				<string>{{ .BinaryPath }}</string>
+			{{ range .Args }}
+				<string>{{ . }}</string>
+			{{ end }}
+			</array>
+			<key>StandardOutPath</key>
+			<string>{{ .StdOutFilePath }}</string>
+			<key>Disabled</key>
+			<false/>
+			<key>RunAtLoad</key>
+			<true/>
+		</dict>
+	</plist>`
+)
+
+// AgentConfig is struct to contain configuration for agent plist file
+type AgentConfig struct {
+	Label          string
+	BinaryPath     string
+	StdOutFilePath string
+	Args           []string
+}
+
+// CreatePlist creates a launchd agent plist config file
+func CreatePlist(config AgentConfig, plistPath string) error {
+	var plistContent bytes.Buffer
+	t, err := template.New("plist").Parse(plistTemplate)
+	if err != nil {
+		return err
+	}
+	err = t.Execute(&plistContent, config)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(plistPath, plistContent.Bytes(), 0644)
+	return err
+}
+
+// LoadPlist loads a launchd agents' plist file
+func LoadPlist(plistFilePath string) error {
+	_, err := exec.Command("launchctl", "load", plistFilePath).Output() // #nosec G204
+	return err
+}
+
+// StartAgent starts a launchd agent
+func StartAgent(label string) error {
+	_, err := exec.Command("launchctl", "start", label).Output() // #nosec G204
```

Comment:
One additional comment, launchctl man page says:

> Start the specified job by label. The expected use of this sub-command is for debugging and testing so that one can manually kick-start an on-demand server.

so maybe it's not really useful to call this?

Reply:
So in case of when we update the tray binary, we forcefully want to start the tray so that the new version is ran..